### PR TITLE
[TRN-191] Add example project for evm/use-hardhat

### PR DESCRIPTION
**Pull request by vldolot**
## Summary

- example project using hardhat deploy
- [TRN-191](https://centralitydev.atlassian.net/browse/TRN-191)

## Checklist

- [x] Add description
- [x] Tag related issue(s)

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -0,0 +1,2052 @@
+# This file is generated by running "yarn install" inside your project.
```
Comment (vldolot) [inline yarn.lock:1]: Are we supposed to include a `yarn.lock` (or package-lock.json) file into this repo?

Comment (ken-centrality) [reply yarn.lock:1]: we use `pnpm` for this repo @vldolot

Comment (vldolot) [reply yarn.lock:1]: Oh okay, thanks. I'll remove it then.

File: examples/evm/use-hardhat/hardhat.config.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    settings: {
+      optimizer: {
+        enabled: true,
+        runs: 200,
+      },
+    },
+  },
+  networks: {
+    porcini: {
+      url: "https://porcini.rootnet.app/",
```
Comment (ken-centrality) [inline examples/evm/use-hardhat/hardhat.config.js:19]: use "@therootnetwork/api" and use `getPublicProviderUrl('porcini')` to get porcini URL

File: examples/evm/use-hardhat/README.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
+# Using `hardhat deploy`
+
+This project demonstrates a basic smart contract deployment and verification to TRN Porcini using `hardhat deploy` plugin. It comes with a sample contract, a test for that contract, and a script that deploys that contract.
+
+## Procedure:
+
+1.  Install the necessary packages and Hardhat plugins.
+
+```
+yarn install
```
Comment (ken-centrality) [inline examples/evm/use-hardhat/README.md:10]:
```suggestion
pnpm install
```

File: examples/evm/use-hardhat/README.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+```
+
+2.  Create an `.env` file and include your private key(s).
```
Comment (ken-centrality) [inline examples/evm/use-hardhat/README.md:13]: use `export` command on terminal, then we can remove the need of `dotenv`
```
export DEPLOYER_PRIVATE_KEY=
export DEPLOYER_ADDRESS=
```

File: examples/evm/use-hardhat/hardhat.config.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
+require("@nomicfoundation/hardhat-ethers");
+require("@nomicfoundation/hardhat-toolbox");
+require("hardhat-deploy");
+require("dotenv").config();
```
Comment (ken-centrality) [inline examples/evm/use-hardhat/hardhat.config.js:4]: swap this with `envalid` library to verify the if `DEPLOYER_PK` and `DEPLOYER` exist

**Review by ken-centrality — COMMENTED** [review]
Make sure you use `pnpm` as command everywhere

File: examples/evm/use-hardhat/hardhat.config.js (hunk 2 above)
Comment (ken-centrality) [reply examples/evm/use-hardhat/hardhat.config.js:19]:
```
const { getPublicProviderUrl } = require("@therootnetwork/api");
```

Comment (vldolot) [reply examples/evm/use-hardhat/hardhat.config.js:19]: So by using `getPublicProviderUrl`, we're gonna be using the `"https://porcini.rootnet.app/archive"` and not `"https://porcini.rootnet.app/"` ?

Comment (vldolot) [reply examples/evm/use-hardhat/hardhat.config.js]: Also, do we have a similar function that provides the `chainId` based on provided network?

Comment (ken-centrality) [reply examples/evm/use-hardhat/hardhat.config.js]: unlike the url, the `chainId` is unlikely to change so i don't mind it to be hard-coded

**Review by ken-centrality — APPROVED** [review]
LGTM! Assuming you have tested the actual deployment @vldolot ?

**Comment by vldolot** [comment]
> LGTM! Assuming you have tested the actual deployment @vldolot ?

Yup. Thanks

**Pull request merged**